Cringe/Services/PpService.cs

```diff
@@ -25,25 +42,87 @@ public async Task<double> CalculatePp(ScoreBase score)
             {
                 await using var stream =
                     new MemoryStream(await _beatmapService.GetBeatmapBytes(score.BeatmapId), false);
-                using var reader = new StreamReader(stream, true);
 
-                var map = Beatmap.Read(reader);
-                var diff = new DiffCalc().Calc(map, (Mods) score.Mods);
+                var beatmap = new StreamWorkingBeatmap(stream, score.BeatmapId);
+
+                var calculator = new OsuPerformanceCalculator();
+
+                var mods = _ruleset.ConvertFromLegacyMods((LegacyMods) (int) score.Mods).Append(new OsuModClassic());
 
-                var pp = new PPv2(new PPv2Parameters(map, diff, score.Accuracy / 100, score.CountMiss, score.MaxCombo,
-                    (Mods) score.Mods)).Total;
+                var difficultyAttributesCalculator = new OsuDifficultyCalculator(_ruleset.RulesetInfo, beatmap);
+                var difficultyAttributes = (OsuDifficultyAttributes) difficultyAttributesCalculator.Calculate(mods);
+
+                var scoreModel = new ScoreInfo(beatmap.BeatmapInfo, _ruleset.RulesetInfo)
+                {
+                    Statistics = new Dictionary<HitResult, int>
+                    {
+                        [HitResult.Great] = score.Count300,
+                        [HitResult.Ok] = score.Count100,
+                        [HitResult.Meh] = score.Count50,
+                        [HitResult.Miss] = score.CountMiss,
+                        [HitResult.SliderTailHit] = difficultyAttributes.SliderCount,
+                        [HitResult.LargeTickMiss] = 0
+                    },
+                    MaxCombo = score.MaxCombo,
+                    Accuracy = score.Accuracy,
+                };
+
+                var performanceAttributes = await calculator.CalculateAsync(scoreModel, difficultyAttributes, CancellationToken.None);
```

Comment:
There's no reason to run it async, replace with `Calculate`
![image](https://github.com/user-attachments/assets/b1a50a75-78bb-4328-8d8c-e307db309c1b)